src/libraries/System.Threading.Tasks.Extensions/tests/ValueTaskTests.cs

```diff
@@ -1278,7 +1278,7 @@ public void NonGeneric_TornRead_DoesNotCrashOrHang()
 
             // _obj is a Task but other fields are from result construction
             vtBoxed = new ValueTask();
-            vtBoxed.GetType().GetField("_obj", BindingFlags.NonPublic | BindingFlags.Instance).SetValue(vtBoxed, Task.CompletedTask);
+            typeof(ValueTask).GetField("_obj", BindingFlags.NonPublic | BindingFlags.Instance).SetValue(vtBoxed, Task.CompletedTask);
```

Comment:
Another potential candidate for better data flow.
We know the static type of the variable here - maybe we should use that to try to resolve the field, and only if that fails produce a warning. But it's a bit weird.

Reply:
The static type of the variable is object, which is why it doesn't work (it would work if it was typed as ValueTask, because it's struct and therefore sealed). We'd need to track the types of things stored here. Analysis in .NET Native did it. I don't know if we really want to go there again.

Reply:
Sorry - I missed that. You're right, I don't think we should do this.